crates/bevy_core/src/name.rs

```diff
@@ -20,9 +24,16 @@ use bevy_reflect::{ReflectDeserialize, ReflectSerialize};
 /// [`Name`] should not be treated as a globally unique identifier for entities,
 /// as multiple entities can have the same name.  [`Entity`] should be
 /// used instead as the default unique identifier.
-#[derive(Reflect, Component, Clone)]
-#[reflect(Component, Default, Debug)]
-#[cfg_attr(feature = "serialize", reflect(Serialize, Deserialize))]
+#[derive(Component, Clone)]
+#[cfg_attr(
+    feature = "bevy_reflect",
+    derive(Reflect),
+    reflect(Component, Default, Debug)
+)]
+#[cfg_attr(
+    all(feature = "serialize", feature = "bevy_reflect"),
+    reflect(Deserialize, Serialize)
+)]
 pub struct Name {
```

Comment:
is it still (de)serializable without reflect?

Reply:
> still

Other crates e.g. `bevy_math` has a separate clause for serialize: https://github.com/bevyengine/bevy/blob/main/crates/bevy_math/src/direction.rs#L88

This crate didn't have a `derive(Serialize, Deserialize)`, but `reflect(Serialize, Deserialize)` only makes sense when having both features available (I think).

I'm guessing not having the separate serde clause is an oversight- should I just add that to this case too?

Reply:
I believe it's implemented manually in a separate module

Reply:
Ah wait, it should be fine as-is.

https://github.com/bevyengine/bevy/blob/main/crates/bevy_core/src/serde.rs

The serde impl is here

Reply:
> I believe it's implemented manually in a separate module

Beat me to it :+1: